airflow/contrib/hooks/aws_athena_hook.py

```diff
@@ -127,14 +127,21 @@ def poll_query_status(self, query_execution_id, max_tries=None):
             elif query_state in self.INTERMEDIATE_STATES:
                 self.log.info('Trial {try_number}: Query is still in an intermediate state - {state}'
                               .format(try_number=try_number, state=query_state))
+            elif query_state in self.FAILURE_STATES:
+                self.log.error('Trial {try_number}: Query is failed - {state}'
+                               .format(try_number=try_number, state=query_state))
+                raise Exception('Athena job {state}.'.format(state=query_state))
             else:
                 self.log.info('Trial {try_number}: Query execution completed. Final state is {state}'
                               .format(try_number=try_number, state=query_state))
                 final_query_state = query_state
                 break
```

Comment:
Hi @bryanyang0528 . I had another check. I don't think we need to add exception here. 

This function `poll_query_status()` helps "_Poll the status of submitted athena query until query state reaches final state_". If you check the current implementation: if the `query_state` is in `self.FAILURE_STATES`, it will go to `else` block, in that block the final `query_state` ( one of `self.FAILURE_STATES`) will be assigned to `final_query_state`, then the while-loop will be stopped by `break`. Eventually, `final_query_state` will be returned.

But your change will break this logic. If the query_state is one of `self.FAILURE_STATES`, we're not expecting an Exception here. Instead, we expect the query_state to be returned (no matter if it's one of `self.FAILURE_STATES`)

Reply:
https://github.com/apache/airflow/blob/17eb94f723502d8bbc716763a7513db0a8afa52d/airflow/contrib/hooks/bigquery_hook.py#L1248
I take `bigquery_hook` as a reference. If the query failed and return the state to Athena Operator, the query job won't be failed and will trigger the downstream job. That's why I would like to add an Exception if the query failed.

Instead of breaking the logic of `pull_query_status()`, I can add the exception in the `AWSAthenaHook`. Which one will you suggest?

Reply:
IMO `poll_query_status` is doing its job nicely (return the final state. As long as it's "final", no matter success or failed). I don't see issue in it.

It should be `poll_query_status`'s downstream's responsibility to behave according to the result from `poll_query_status`.

Reply:
OK, I'll add those exceptions to `AthenaOperator`. Thx.

Reply:
I'll close this PR and open a new one.